lib/become_web/controllers/page_controller.ex

```diff
@@ -1,7 +1,74 @@
 defmodule BecomeWeb.PageController do
   use BecomeWeb, :controller
+  alias Become.Finances.File , as: FinanceFile
 
   def index(conn, _params) do
-    render(conn, "index.html")
+    changeset = FinanceFile.changeset(%FinanceFile{}, %{})
+    render(conn, "index.html", changeset: changeset)
   end
+
+  def convert(conn, %{"file" => %{"path" => upload}}) do
+    csv = decode_csv(upload.path)
+    [title | content] = csv
+    conversion_rate = get_exchange_rate()
+    products = get_product_with_currency(content, conversion_rate)
+    render(conn, "convert.html", products: products)
+  end
+
+  def decode_csv(path) do
+    File.stream!(path) 
+    |> CSV.decode 
+    |> Enum.map(fn {:ok, [name, price]} -> [name,price]  end)
+  end
+
+  def get_exchange_rate() do
+    country_rate = get_country_exchange_rate()
+    bitcoin_rate = get_bitcoin_exchange_rate()
+    Map.merge(country_rate, bitcoin_rate)
+    
+  end
+
+  def get_country_exchange_rate() do
+    {:ok, %HTTPoison.Response{status_code: 200, body: body}} = HTTPoison.get("https://api.exchangerate-api.com/v4/latest/MYR")
```

Comment:
This call assumes network and third party API is 100% available and does not handle errors for it not being available. An {:error, error} tuple being handled here would be nice.